### PR TITLE
name change csvParsers to csvDateFormat

### PR DESCRIPTION
**Pull request by sgilic**
1. Name change csvParsers to csvDateFormat:
AstInternal
ParserInternal
Pretty
Tests

behaviour change: csv date format will now apply to all date columns globaly.
e.g: instaed of "b=ISO8206" -> "ISO8206"

2. Fixed warnings at parser internal function body

### CODE REVIEW

File: hssqlppp/src/Database/HsSqlPpp/Pretty.lhs [hunk 1]
```diff
@@ -1140,15 +1140,15 @@ syntax maybe should error instead of silently breaking
 
 > ppCsvOpts :: CsvOptions -> Doc
 > ppCsvOpts NewCsvOptions
->   { csvFilePath = path
+>   { csvFilePath = _path
```
Comment (soupi) [inline hssqlppp/src/Database/HsSqlPpp/Pretty.lhs]: you can use `{..}` here instead and then use the full name of each field

File: hssqlppp/tests/Database/HsSqlPpp/Tests/Parsing/Dml.lhs [hunk 2]
```diff
@@ -233,11 +233,11 @@ copy, bit crap at the moment
 >            , csvErrorOptions = Just EOAbort
 >            , csvLimit = Just 10
 >            , csvOffset = Just 1
->            , csvParsers = Just "b=oracle"
+>            , csvDateFormat = Just "b=oracle"
```
Comment (soupi) [inline hssqlppp/tests/Database/HsSqlPpp/Tests/Parsing/Dml.lhs]: this also needs to be changed

File: hssqlppp/tests/Database/HsSqlPpp/Tests/Parsing/Dml.lhs [hunk 3]
```diff
@@ -233,11 +233,11 @@ copy, bit crap at the moment
...
 >            }
 >          )
 >        ]
->      , s "copy tbl (a,b) from 'filepath' with options field delimiter '|' text qualifier '?' null marker 'null' on error skip row 100 offset 1 limit 10 parsers 'b=oracle';"
+>      , s "copy tbl (a,b) from 'filepath' with options field delimiter '|' text qualifier '?' null marker 'null' on error skip row 100 offset 1 limit 10 date format 'b=oracle';"
```
Comment (soupi) [inline hssqlppp/tests/Database/HsSqlPpp/Tests/Parsing/Dml.lhs]: and this

File: hssqlppp/tests/Database/HsSqlPpp/Tests/Parsing/Dml.lhs [hunk 4]
```diff
@@ -250,7 +250,7 @@ copy, bit crap at the moment
 >            , csvErrorOptions = Just (EOSkipRowLimit 100 NoReportSkippedRows)
 >            , csvLimit = Just 10
 >            , csvOffset = Just 1
->            , csvParsers = Just "b=oracle"
+>            , csvDateFormat = Just "b=oracle"
```
Comment (soupi) [inline hssqlppp/tests/Database/HsSqlPpp/Tests/Parsing/Dml.lhs]: and this

File: hssqlppp/src/Database/HsSqlPpp/Pretty.lhs (hunk 1 above)
Comment (sgilic) [reply hssqlppp/src/Database/HsSqlPpp/Pretty.lhs]: fixed it

File: hssqlppp/tests/Database/HsSqlPpp/Tests/Parsing/Dml.lhs (hunk 2 above)
Comment (sgilic) [reply hssqlppp/tests/Database/HsSqlPpp/Tests/Parsing/Dml.lhs]: fixed it

File: hssqlppp/tests/Database/HsSqlPpp/Tests/Parsing/Dml.lhs (hunk 3 above)
Comment (sgilic) [reply hssqlppp/tests/Database/HsSqlPpp/Tests/Parsing/Dml.lhs]: fixed it

File: hssqlppp/tests/Database/HsSqlPpp/Tests/Parsing/Dml.lhs (hunk 4 above)
Comment (sgilic) [reply hssqlppp/tests/Database/HsSqlPpp/Tests/Parsing/Dml.lhs]: fixed it

**Pull request merged**